### PR TITLE
tai/t12-refactor-no-hardcode

### PR DESCRIPTION
**Pull request by ctlai95**
- all sizing depend on `unit` variable
- new point class to store x y coordinates
- tetromino pieces are generated based on an origin point
- tetromino pieces are now stored in a list
- piece movement and drawing are now called recursively

### CODE REVIEW

**Comment by henrytran28** [comment]
Now that I think about it the point and square class seem to be unnecessary

**Comment by henrytran28** [comment]
I've been reading up a bit on the design https://stackoverflow.com/questions/1863281/philosophical-design-questions-for-oop-tetris

We could probably separate into 3-4 different files. I could do this if you want. 
Something like the main file (subject), movement, board, pieces as observers.

**Comment by ctlai95** [comment]
You're right that the point class isn't really necessary, but the reason I included it is to be clear that there is an x and y coordinate when defining a tetromino's origin. And I think it's also nice to separate the square class because of the way OpenGL draws the squares (as two triangles), otherwise the tetromino class would get too messy imo.

File: python-tetris.py [hunk 1]
```diff
@@ -120,28 +70,17 @@ def on_text_motion(motion):
...
-                                 ('v2i', current_piece.right.position()))
-
-    pyglet.graphics.draw_indexed(4, pyglet.gl.GL_TRIANGLES,
-                                 [0, 1, 2, 0, 2, 3],
-                                 ('v2i', current_piece.top.position()))
-
-    pyglet.graphics.draw_indexed(4, pyglet.gl.GL_TRIANGLES,
-                                 [0, 1, 2, 0, 2, 3],
-                                 ('v2i', current_piece.bottom.position()))
+    for i in range(0, current_piece.size):
```
Comment (henrytran28) [inline python-tetris.py]: for i in range(current_piece.size)

**Comment by henrytran28** [comment]
Looking good buddy.

File: python-tetris.py [hunk 2]
```diff
@@ -1,111 +1,61 @@
...
+        for c in self.coords:
+            t.append(
+                (c[0], c[1],
+                 c[0]+UNIT, c[1],
+                 c[0]+UNIT, c[1]+UNIT,
+                 c[0], c[1]+UNIT)
+            )
+        return t
 
     def move_left(self):
```
Comment (henrytran28) [inline python-tetris.py]: Can you make all the move functions to one function and just accept a direction argument?

Comment (ctlai95) [inline python-tetris.py]: Good idea 👍

**Comment by ctlai95** [comment]
I hate you @henrytran28

**Pull request merged**